tools/reviewer/cas_proxy/Proxy.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2018 The StartupOS Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.startupos.tools.reviewer.cas_proxy;
+
+import com.google.common.hash.Hashing;
+import com.google.protobuf.ByteString;
+import com.google.startupos.common.firestore.FirestoreProtoClient;
+import com.google.startupos.tools.reviewer.proxy.AuthProtos.FileDownloadRequest;
+import com.google.startupos.tools.reviewer.proxy.AuthProtos.FileDownloadResponse;
+import com.google.startupos.tools.reviewer.proxy.AuthProtos.FileUploadRequest;
+import com.google.startupos.tools.reviewer.proxy.AuthProtos.FileUploadResponse;
+import com.google.startupos.tools.reviewer.proxy.CASProxyServiceGrpc;
+
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import io.grpc.stub.StreamObserver;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class Proxy {
+
+  // TODO(vmax): make bucket configurable
+  public static final String CAS_BUCKET_NAME = "test-bucket-cas";
+
+  static class AccessDeniedException extends Exception {}
+
+  static class Service extends CASProxyServiceGrpc.CASProxyServiceImplBase {
```

Comment:
Let's rename `Service->CasService`.

Reply:
@oferb fixed in https://github.com/google/startup-os/pull/541/commits/881572de2fce8e235dd42d5a74e171b80f38636c